griptape/artifacts/base_media_artifact.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+from abc import ABC
+
+import string
+import time
+import random
+from typing import Optional
+
+from attr import define, field
+from griptape.artifacts import BlobArtifact
+import base64
+
+
+@define
+class MediaArtifact(BlobArtifact):
+    """MediaArtifact is a type of BlobArtifact that represents media (image, audio, video, etc.)
+    and can be extended to support a specific media type.
+
+    Attributes:
+        value: Raw bytes representing media data.
+        name: Artifact name, generated using creation time and a random string.
+        mime_type: The mime type of the image, like image/png or audio/wav.
+        model: Optionally specify the model used to generate the media.
+        prompt: Optionally specify the prompt used to generate the media.
+    """
+
+    artifact_type: str = field(default="media", kw_only=True, metadata={"serializable": True})
+    format: str = field(kw_only=True, metadata={"serializable": True})
```

Comment:
Do these need to be split out like this? Do they serve any purpose other than building `mime_type`? I think `artifact_type` is a bit confusing because it's really the media type.

Reply:
These attributes contribute to `mime_type` as well as artifact name. Independent `format` metadata in particular is useful for things like enforcing format limitations on model inputs.